### PR TITLE
Add '-l' option to all mqtt commands to activate logfile logging

### PR DESCRIPTION
**Pull request by gitseti**
**Motivation**
- Currently the cli, logs all commands to ~/.mqtt-cli/logs
- This behavior is unexpected and should be manually activated

**Changes**
- Add '-l' option to manually activate the logging of all cli commands

### CODE REVIEW

File: src/main/java/com/hivemq/cli/commands/cli/PublishCommand.java [hunk 1]
```diff
@@ -103,6 +104,9 @@ public PublishCommand(final @NotNull MqttClientExecutor mqttClientExecutor) {
     @CommandLine.Option(names = {"-up", "--userProperty"}, converter = Mqtt5UserPropertyConverter.class, description = "A user property of the publish message", order = 1)
     @Nullable private Mqtt5UserProperty[] userProperties;
 
+    @CommandLine.Option(names = {"-l"}, defaultValue = "false", description = "Log to ~./mqtt.cli/logs (Configurable through ~/.mqtt-cli/config.properties)", order = 1)
```
Comment (sauroter) [inline src/main/java/com/hivemq/cli/commands/cli/PublishCommand.java:107]:
```suggestion
    @CommandLine.Option(names = {"-l"}, defaultValue = "false", description = "Log to ~/.mqtt.cli/logs (Configurable through ~/.mqtt-cli/config.properties)", order = 1)
```

File: src/main/java/com/hivemq/cli/commands/cli/SubscribeCommand.java [hunk 2]
```diff
@@ -102,6 +103,9 @@ public SubscribeCommand(final @NotNull MqttClientExecutor mqttClientExecutor,
     @CommandLine.Option(names = {"-T", "--showTopics"}, defaultValue = "false", description = "Prepend the specific topic name to the received publish", order = 1)
     private boolean showTopics;
 
+    @CommandLine.Option(names = {"-l"}, defaultValue = "false", description = "Log to ~./mqtt.cli/logs (Configurable through ~/.mqtt-cli/config.properties)", order = 1)
```
Comment (sauroter) [inline src/main/java/com/hivemq/cli/commands/cli/SubscribeCommand.java:106]:
```suggestion
    @CommandLine.Option(names = {"-l"}, defaultValue = "false", description = "Log to ~/.mqtt.cli/logs (Configurable through ~/.mqtt-cli/config.properties)", order = 1)
```

File: src/main/java/com/hivemq/cli/commands/cli/PublishCommand.java (hunk 1 above)
Comment (sauroter) [reply src/main/java/com/hivemq/cli/commands/cli/PublishCommand.java:107]: I would suggest to replace "~" with "$HOME"

File: src/main/java/com/hivemq/cli/commands/cli/TestBrokerCommand.java [hunk 3]
```diff
@@ -71,6 +72,9 @@
     @CommandLine.Option(names = {"-q", "--qosTries"}, defaultValue = "10", description = "The amount of publishes to send to the broker on every qos level", order = 1)
     private @NotNull Integer qosTries;
 
+    @CommandLine.Option(names = {"-l"}, defaultValue = "false", description = "Log to ~./mqtt.cli/logs (Configurable through ~/.mqtt-cli/config.properties)", order = 1)
```
Comment (sauroter) [inline src/main/java/com/hivemq/cli/commands/cli/TestBrokerCommand.java:75]:
```suggestion
    @CommandLine.Option(names = {"-l"}, defaultValue = "false", description = "Log to ~/.mqtt.cli/logs (Configurable through ~/.mqtt-cli/config.properties)", order = 1)
```

File: src/main/java/com/hivemq/cli/commands/shell/ShellCommand.java [hunk 4]
```diff
@@ -90,6 +90,9 @@
     @CommandLine.Option(names = {"--help", "-h"}, usageHelp = true, description = "display this help message")
     boolean usageHelpRequested;
 
+    @CommandLine.Option(names = {"-l"}, defaultValue = "false", description = "Log to ~./mqtt.cli/logs (Configurable through ~/.mqtt-cli/config.properties)", order = 1)
```
Comment (sauroter) [inline src/main/java/com/hivemq/cli/commands/shell/ShellCommand.java:93]:
```suggestion
    @CommandLine.Option(names = {"-l"}, defaultValue = "false", description = "Log to ~/.mqtt.cli/logs (Configurable through ~/.mqtt-cli/config.properties)", order = 1)
```

File: src/main/java/com/hivemq/cli/commands/cli/PublishCommand.java [hunk 5]
```diff
@@ -115,7 +119,13 @@ public void run() {
             if (isVerbose()) put("writer1.level", "trace");
         }};
 
-        LoggerUtils.useDefaultLogging(configurationMap);
+        if (logToLogfile) {
```
Comment (schaebo) [inline src/main/java/com/hivemq/cli/commands/cli/PublishCommand.java:122]: This is duplicated and could be moved to a util. e.g. LoggerUtils.setupLogging(logToLogfile)

**Pull request merged**